proposed/http-factory/http-factory.md

```diff
@@ -144,6 +144,9 @@ interface StreamFactoryInterface
      * @param string $filename
      * @param string $mode
      *
+     * @throws \InvalidArgumentException
+     *  If the file cannot be opened using the mode requested.
```

Comment:
Do we want to limit this to `InvalidArgumentException`? I can think of a few other candidates:

- `DomainException`, as it's related to the HTTP message _domain_ context.
- `UnexpectedValueException`, as we are getting a `false` return value from `fopen()`, and not the expected resource.
- `RuntimeException`, as it's happening during runtime.

If we choose _not_ to define it, the implementation can then choose what to do in such scenarios, including what type of exception to throw. Thoughts?

Reply:
Probably makes sense to use `Exception` here, or define a specific exception interface.

Reply:
Defining a specific exception interface would require going back to Draft status; indicating simply `Exception` would be a clarification. How would you like to proceed?

Reply:
Hmm, reading through it again, I actually think that `InvalidArgumentException` is the right type because it means that either given `$filename` or `$mode` is invalid.

I think `Exception` is probably too loose, `DomainException` is awkward (I only use it for application code), `UnexpectedValueException` doesn't make sense because the parameters are have strict types, and `RuntimeException` is more vague than the current value.

Reply:
Now that I say that, I think this is a good point https://github.com/php-fig/fig-standards/pull/904#issuecomment-300562216. Perhaps `UnexpectedValueException` is the better type due to the fact that `InvalidArgumentException` is related to type more than `UnexpectedValueException`.

Reply:
RuntimeException is the correct exception to use here. The argument may be correct, but somebody might deleted the file; therefore, InvalidArgumentException is not correct. The InvalidArgumentException should be thrown if $mode is invalid:

`````
 /**
  * ...
  * @throws \RuntimeException If the file cannot be opened.
  * @throws \InvalidArgumentException If the $mode is invalid.
  */
`````

Reply:
@jkufner smart, somehow I didn't even consider throwing two different exception types. Makes perfect sense to me. How about you @weierophinney ?

Reply:
I've checked with the core committee, and review status is considered less RC and more beta under the current by-laws.

Additionally, I'd consider this a clarification: it's clarifying intended behavior by implementors to consumers when (a) the mode is incorrect, and/or (b) the file cannot be opened.

So: make the change, and I'll review one last time for merge.

Reply:
Updated.